src/main/java/emissary/core/BaseDataObject.java

```diff
@@ -373,7 +373,15 @@ public SeekableByteChannelFactory getChannelFactory() {
                 return theData;
             case CHANNEL_ONLY:
                 try (final SeekableByteChannel sbc = this.seekableByteChannelFactory.create()) {
-                    final int buffSize = (int) Math.min(sbc.size(), Integer.MAX_VALUE);
+                    final long channelSize = sbc.size();
+                    final long truncatedBy = channelSize - Integer.MAX_VALUE;
+                    int buffSize = 0;
+                    if (truncatedBy > 0) {
+                        logger.warn("Data for [{}] truncated by {} bytes due to size constraints", shortName(), truncatedBy);
```

Comment:
Clarify this to indicate that the **returned** data was truncated; the current verbiage suggests that the backing data may have been truncated, which is inaccurate.